src/board/system76/common/battery.c

```diff
@@ -75,6 +75,22 @@ int16_t battery_charger_configure(void) {
     return battery_charger_disable();
 }
 
+void battery_limit_capacity(void) {
+    // Clevo requires limiting charge to 90% of the SBS reported capacity.
+    battery_info.full_capacity = ((uint32_t)battery_info.full_capacity * 90) / 100;
+
+    // XXX: RemainingCapacity() must *exactly* equal FullChargeCapacity() when
+    // "full" for Linux to report battery as fully charged.
+    battery_info.remaining_capacity = MIN(
+        battery_info.remaining_capacity,
+        battery_info.full_capacity
+    );
+
+    // Manual calculate RSoC based on the new FullChargeCapacity()
+    battery_info.charge = (battery_info.remaining_capacity * 100) / battery_info.full_capacity;
```

Comment:
This is another potential overflow point

Reply:
Done